File: python/nt_toolbox/plot_levelset.py
import numpy as np
import matplotlib.pyplot as plt
from nt_toolbox.signal import imageplot

def plot_levelset(Z, level=0, f=[]):
    """
        f is supposed to be of the same shape as Z
    """
    if len(f) == 0:
        f = np.copy(Z)
        
    n,p = np.shape(Z)
    X,Y = np.meshgrid(np.arange(0,n),np.arange(0,p))
    plt.contour(X, Y, Z,[level],linewidths=2, colors="red")
    imageplot(f)

File: r/nt_toolbox/plot_levelset.R




plot_levelset <- function(Z, f=c(), title="", lw=1.5, sbpt=c()){
  ####
  # f is supposed to be of the same shape as Z
  ####
  if (length(f)==0){
    f <- Z }

  Z <- as.cimg(t(Z))
  ct <- contours(Z, nlevels=1)
  imageplot(f, title, sbpt)
  purrr::walk(ct,function(v) lines(v$x,v$y,col="red",lw=lw))
}